compile_docker.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+
+BASE_IMAGE="ubuntu:jammy-20220421"
+
+IMAGE_NAME="scons_image"
+CONTAINER_NAME="scons_container"
+
+COMPILER_DIR=embedded/platforms/toolchain-gccarmnoneeabi/
+
+# wrapper variables 
+COMPONENT=""
+
+# pull base image 
+docker pull ${BASE_IMAGE}
+	
+# build base ubuntu image  
+docker build -t ${IMAGE_NAME} .
+
+docker-compose up -d
+
+# download compiler 
+
+# search recursively through repo dir for compiler 
+find . -name "gcc-arm-11.2-2022.02-x86_64-arm-none-eabi*" > compiler_exist.txt
+
+if [ -s compiler_exist.txt ]; then
+
+	echo "Compiler files present. Skipping download"
+else
+	echo "No compiler files found. Downloading."
+	docker exec ${CONTAINER_NAME} wget https://developer.arm.com/-/media/Files/downloads/gnu/11.2-2022.02/binrel/gcc-arm-11.2-2022.02-x86_64-arm-none-eabi.tar.xz
```

Comment:
why are you doing this in the context of the container? Just execute the wget command from the shell script, no need to involve the docker container

Reply:
I was thinking of windows users or linux/mac users that dont have wget binary - since the container does I just ran the cmd in the container. if were making it a requirement to have WSL we can also say install wget

Reply:
that's a good idea but I don't think this shell script will run on windows regardless, only on linux/mac. Might be a good idea to test with cygwin on windows at some point.